pubnub-chat-api/src/commonMain/kotlin/com/pubnub/chat/Chat.kt

```diff
@@ -136,58 +339,166 @@ interface Chat {
+     * @return [PNFuture] containing [PNPushAddChannelResult]
+     */
     fun registerPushChannels(channels: List<String>): PNFuture<PNPushAddChannelResult>
 
+    /**
+     * Specifies the channel or channels on which a registered device will no longer receive push notifications for new messages.
+     *
+     * @param channels list of [Channel.id]
+     *
+     * @return [PNFuture] containing [PNPushRemoveChannelResult]
+     */
     fun unregisterPushChannels(channels: List<String>): PNFuture<PNPushRemoveChannelResult>
 
+    /**
+     * Disable push notifications for a device on all registered channels.
+     *
+     * @return [PNFuture] containing [PNPushRemoveChannelResult]
+     */
     fun unregisterAllPushChannels(): PNFuture<Unit>
 
-    fun getThreadChannel(message: Message): PNFuture<ThreadChannel>
-
+    /**
+     * Returns info on all messages you didn't read on all joined channels. You can display this number on UI in the
+     * channel list of your chat app.
+     *
+     * @param limit Number of objects to return in response. The default (and maximum) value is 100.
+     * @param page Object used for pagination to define which previous or next result page you want to fetch.
+     * @param filter Expression used to filter the results. Returns only these channels whose properties satisfy the given expression are returned.
+     * @param sort A collection to specify the sort order. Available options are id, name, and updated. Use asc or desc.
+     *
+     * @return [PNFuture] containing set of [GetUnreadMessagesCounts]
+     */
     fun getUnreadMessagesCounts(
         limit: Int? = null,
         page: PNPage? = null,
         filter: String? = null,
         sort: Collection<PNSortKey<PNMembershipKey>> = listOf(),
     ): PNFuture<Set<GetUnreadMessagesCounts>>
 
+    /**
+     * Allows you to mark as read all messages you didn't read on all joined channels.
+     *
+     * @param limit Number of objects to return in response. The default (and maximum) value is 100.
+     * @param page Object used for pagination to define which previous or next result page you want to fetch.
+     * @param filter Expression used to filter the results. Returns only these channels whose properties satisfy the given expression are returned.
+     * @param sort A collection to specify the sort order. Available options are id, name, and updated. Use asc or desc.
+     *
+     * @return [PNFuture] containing set of [MarkAllMessageAsReadResponse]
+     */
     fun markAllMessagesAsRead(
         limit: Int? = null,
         page: PNPage? = null,
         filter: String? = null,
         sort: Collection<PNSortKey<PNMembershipKey>> = listOf(),
     ): PNFuture<MarkAllMessageAsReadResponse>
 
+    /**
+     * Retrieves all channels referenced in the [Channel.sendText] that match the provided 3-letter string from
+     * your app's keyset.
+     * @param text At least a 3-letter string typed in after # with the channel name you want to reference.
```

Comment:
the implementation suggests that `text` should contain the "#"...
in any case, I think we should hide this function until we have MessageDraft as the implementation will change and we don't want to support it indefinitely

same with user suggestions

Reply:
Good point. I will move them to ChatInternal.